### PR TITLE
TE-3769: creating rules for checking if tag of component isn't div

### PR DESCRIPTION
**Pull request by KostKravchenko**
Ticket: https://spryker.atlassian.net/browse/TE-3769

### CODE REVIEW

**Review by supproduction — APPROVED** [review]

**Comment by KostKravchenko** [comment]
Ticket: https://spryker.atlassian.net/browse/TE-3767

File: rules/correct-tag-name.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+const { dim, bold } = require('colors');
+const { Rule } = require('../api');
+
+module.exports = class extends Rule {
+    getName() {
+        return 'correct-tag-name';
```
Comment (tolerants) [inline rules/correct-tag-name.js:7]: Suggest to rename. Or extend this logic to satisfy the rule meaning: tag name is correct if:
- tag is not a div
- tag is one of standard html tags
- tag is a custom element (so it consist dash inside name and the tag name and component name are the same)

File: rules/correct-tag-name.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    test(data) {
+        data.components.forEach(component => {
+            const { twig } = component.files;
+
+            if (twig.exists) {
+                const { definitions } = twig.api.external;
+
+                definitions.forEach(definition => {
+                    if (definition.name !== 'config') {
+                        return
```
Comment (tolerants) [inline rules/correct-tag-name.js]: Please add `;`.

File: rules/correct-tag-name.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+                    }
+
+                    const { contract } = definition;
+                    const tagWordIndex = contract.indexOf('tag:');
+                    const croppedStringFromTagWord = contract.slice(tagWordIndex);
+                    const tagPropertyString = croppedStringFromTagWord.slice(0, croppedStringFromTagWord.indexOf('\n'));
+                    const tagName = tagPropertyString.slice(tagPropertyString.indexOf('\'') + 1, tagPropertyString.lastIndexOf('\''));
+
+                    if (tagName === 'div') {
+                        this.outcome.addError(`it shouldn't be tag property with div value in config of ${component.type} ${bold(component.name)}:\n${dim(component.path)}`);
```
Comment (tolerants) [inline rules/correct-tag-name.js]: Please start sentence with capital letter.

File: rules/correct-tag-name.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+                definitions.forEach(definition => {
+                    if (definition.name !== 'config') {
+                        return
+                    }
+
+                    const { contract } = definition;
+                    const tagWordIndex = contract.indexOf('tag:');
+                    const croppedStringFromTagWord = contract.slice(tagWordIndex);
+                    const tagPropertyString = croppedStringFromTagWord.slice(0, croppedStringFromTagWord.indexOf('\n'));
+                    const tagName = tagPropertyString.slice(tagPropertyString.indexOf('\'') + 1, tagPropertyString.lastIndexOf('\''));
```
Comment (tolerants) [inline rules/correct-tag-name.js]: I'd extract to constants `tagPropertyString.indexOf('\'') + 1` and `tagPropertyString.lastIndexOf('\'')` values to increase code readability.

**Review by tolerants — CHANGES_REQUESTED** [review]

File: rules/correct-tag-name.js (hunk 1 above)
Comment (KostKravchenko) [reply rules/correct-tag-name.js:7]: I extended logic.
I wrote rules to check if tag name is a valid html tag, if there are forbidden symbols in tag name

File: rules/correct-tag-name.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+module.exports = class extends Rule {
+    getName() {
+        return 'correct-tag-name';
+    }
+
+    test(data) {
+        data.components.forEach(component => {
+            const { twig } = component.files;
+
+            if (twig.exists) {
```
Comment (FRITS-Kh) [inline rules/correct-tag-name.js]: Please use early return here.

Comment (FRITS-Kh) [inline rules/correct-tag-name.js]: Please use early return here.

Comment (FRITS-Kh) [inline rules/correct-tag-name.js]: Please use early return here.

**Review by FRITS-Kh — CHANGES_REQUESTED** [review]

**Review by FRITS-Kh — APPROVED** [review]

**Pull request merged**